File: Seminar6/java/src/SRP/Accounting.java
package SRP;

public class Accounting {
    private int baseSalary;
    private double ratio = 0.25;

    public Accounting(int baseSalary) {
        this.baseSalary = baseSalary;
    }

    public int calculateNetSalary() {
        int tax = (int) (baseSalary * ratio);
        return baseSalary - tax;
    }

    public void giveOutSalary(Employee employee) {
        employee.takeSalary(this.calculateNetSalary());
    }
}

File: Seminar6/python/SRP/Accounting.py
class Accounting:
    def __init__(self, base_salary):
        self.base_salary = base_salary
        self.ratio = 0.25

    def calculateNetSalary():
        tax = int(self.base_salary * ratio)
        return self.base_salary - tax

    def giveOutSalary(self, emp: Employee):
        emp.takeSalary(self.calculateNetSalary())